### ISSUE TITLE
how to pass the number of label_names to HF_trainer 

### ISSUE THREAD

**Issue by ianni67** [neuml/txtai#647]
Hi, probably a naive question but I'm a bit loss here.
I'm fine-tuning a HF model with HF_Trainer and my training labels are numeric but not all contiguous, for example, [0,2,4,5,6,9].
I get an error from the train function that I run:
`model, tokenizer = trainer("dlicari/distil-ita-legal-bert", X_train)`
as follows:
`nll_loss_forward_reduce_cuda_kernel_2d: block: [0,0,0], thread: [0,0,0] Assertion `t >= 0 && t < n_classes` failed.`

I could understand that labels should be contiguous, for example, [0,1,2,3,4,5] with no missing values in the dataset, but my data does have missing labels. I would like to directly pass the list of labels to the trainer, such as: 
`model, tokenizer = trainer("dlicari/distil-ita-legal-bert", X_train,  label_names=[0,1,2,3,4,5, 6,7,8,9] ) `

OR, BETTER, to pass only the number of labels to be expected, such as, in the example above, 10.

Is there a way to solve this problem?
Thank you very much

**Comment by davidmezzetti** [comment]
Hello.

The `trainer` pipeline expects training data to be in a columnar format, a little different than `scikit-learn`.

In other words something like this.

```
[{"text", "label"}]
```

Can be a list of dictionaries, a HF dataset or a pandas dataframe for example. 

Check out this notebook for more: https://colab.research.google.com/github/neuml/txtai/blob/master/examples/16_Train_a_text_labeler.ipynb

**Comment by davidmezzetti** [comment]
Closing this issue due to inactivity. Please re-open or open a new issue if there are further questions.
